src/main/java/com/bookMyShow/bookMyShow/services/SeatServiceImpl.java

```diff
@@ -1,47 +1,73 @@
 package com.bookMyShow.bookMyShow.services;
 
-import com.bookMyShow.bookMyShow.constants.ApiConstant;
-import com.bookMyShow.bookMyShow.constants.AuditoriumConstant;
-import com.bookMyShow.bookMyShow.exceptions.Error;
+import com.bookMyShow.bookMyShow.Dto.SeatDto;
+import com.bookMyShow.bookMyShow.exceptions.ElementAlreadyExistsException;
+import com.bookMyShow.bookMyShow.exceptions.ElementNotFoundException;
 import com.bookMyShow.bookMyShow.models.Auditorium;
 import com.bookMyShow.bookMyShow.models.Seat;
-import com.bookMyShow.bookMyShow.models.SeatType;
 import com.bookMyShow.bookMyShow.repositories.AuditoriumRepository;
 import com.bookMyShow.bookMyShow.repositories.SeatRepository;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class SeatServiceImpl implements SeatService {
     @Autowired
-    SeatRepository seatRepository;
+    private SeatRepository seatRepository;
 
     @Autowired
-    AuditoriumRepository auditoriumRepository;
+    private AuditoriumRepository auditoriumRepository;
 
     @Override
-    public Seat save(String seatNumber, SeatType seatType, Long auditoriumId) {
+    public List<Seat> createSeat(Long auditoriumId,List<SeatDto> seatDetailsDto) {
         Optional<Auditorium> auditorium = auditoriumRepository.findById(auditoriumId);
 
-        if(!auditorium.isPresent()) {
-            Error error = Error.builder()
-                    .code(HttpStatus.NOT_FOUND)
-                    .status(ApiConstant.ERROR)
-                    .message(AuditoriumConstant.AUDITORIUM_NOT_FOUND)
-                    .build();
-
-            throw error;
+        if (auditorium.isEmpty()) {
+            throw new ElementNotFoundException("Auditorium not found");
         }
 
-        Seat seat = new Seat();
-        seat.setSeatNumber(seatNumber);
-        seat.setSeatType(seatType);
-        seat.setAuditorium(auditorium.get());
+        List<String> seatNumbers = seatDetailsDto.stream()
+                .map(SeatDto::getSeatNumber)
+                .collect(Collectors.toList());
+
+
+        List<Seat> seats = seatRepository.findByAuditoriumIdAndSeatNumberIn(
+                auditoriumId,
+                seatNumbers
+        );
+
+        Map<String, Seat> existingSeatMap = seats.stream()
+                .collect(Collectors.toMap(
+                        seat -> seat.getSeatNumber() + "-" + seat.getRow(),
+                        seat -> seat
+                ));
+
+
+        List<Seat> seatDtos = new ArrayList<>();
```

Comment:
This validation is not needed.  You can do it if you want, but it is not needed.